pkg/database/vercode.go

```diff
@@ -341,6 +281,79 @@ func (db *Database) DeleteVerificationCode(code string) error {
 		Error
 }
 
+// UpdateStats increments VerificationCode statistics incrementing stats but the number issued.
+func (db *Database) UpdateStats(ctx context.Context, codes ...*VerificationCode) {
+	issued := len(codes)
+	if issued == 0 {
+		return
+	}
+	v := codes[0]
+
+	date := timeutils.Midnight(v.CreatedAt)
+	var merr *multierror.Error
+
+	// If the issuer was a user, update the user stats for the day.
+	if v.IssuingUserID != 0 {
+		sql := fmt.Sprintf(`
+			INSERT INTO user_stats (date, realm_id, user_id, codes_issued)
+				VALUES ($1, $2, $3, 1)
+			ON CONFLICT (date, realm_id, user_id) DO UPDATE
+				SET codes_issued = user_stats.codes_issued + %d
+		`, issued)
+
+		if err := db.db.Exec(sql, date, v.RealmID, v.IssuingUserID).Error; err != nil {
+			multierror.Append(merr, fmt.Errorf("failed to update stats: %v", err))
```

Comment:
I don't know if multierror is necessary here. I'd suggest we pull the logger out of the context at the beginning and just call `logger.Warnw` when appropriate.

Reply:
@whaught FYI

Reply:
Sure. Will fold into an unrelated PR